core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -717,6 +722,13 @@ public function cut($blnDoNotRedirect=false)
 			$this->set['ptable'] = $this->ptable;
 		}
 
+		$this->loadActiveRecord();
+
+		$this->denyAccessUnlessGranted(
+			ContaoCorePermissions::DC_PREFIX . $this->strTable,
+			new UpdateAction($this->strTable, $this->activeRecord->row(), array_merge($this->activeRecord->row(), $this->set))
```

Comment:
The new data should probably not contain the full record, since only the written data should be relevant to the permission check?

Reply:
No it should of course contain everything, that's why everythign is loaded.